### ISSUE TITLE
fix(vscode): toImportSpecifier() prepends current Windows drive to POSIX absolute paths

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3836]
- Markuplint version: `4.18.1` (v4 branch HEAD)
- Parser lang: N/A (vscode extension server)
- Node.js version: `24` (CI matrix)
- OS: Windows

## Describe the bug

`vscode/src/server/get-module.ts`'s `toImportSpecifier()` prepends the current Windows drive letter to POSIX absolute paths. When called on a Windows runtime with `/tmp/foo.mjs`, it returns `'file:///D:/tmp/foo.mjs'` instead of the expected `'file:///tmp/foo.mjs'`. POSIX paths must round-trip unchanged because the VS Code extension can receive POSIX paths from a Windows process via WSL or remote development scenarios.

The unit tests at `vscode/src/server/get-module.spec.ts:39` and `:43` already encode this contract, and they fail on Windows CI.

## Code Example or Playground URL

```ts
// vscode/src/server/get-module.spec.ts
test('converts a POSIX absolute path to a file:// URL', () => {
  expect(toImportSpecifier('/tmp/markuplint/lib/index.mjs'))
    .toBe('file:///tmp/markuplint/lib/index.mjs');
});

test('URL-encodes spaces in POSIX paths', () => {
  expect(toImportSpecifier('/tmp/my project/lib.mjs'))
    .toBe('file:///tmp/my%20project/lib.mjs');
});
```

## Steps To Reproduce

1. `git checkout v4`
2. Run the test on Windows: `npx vitest run vscode/src/server/get-module.spec.ts`
3. Both POSIX-path tests fail with received `'file:///D:/tmp/...'`

## The current behavior

```
AssertionError: expected 'file:///D:/tmp/markuplint/lib/index.mjs'
                to be 'file:///tmp/markuplint/lib/index.mjs'
```

The implementation likely uses `pathToFileURL()` (or equivalent), which on Windows resolves `/tmp/...` against the current drive (`D:` on the GitHub-hosted Windows runner) and prepends the drive letter.

## The expected behavior

POSIX absolute paths must round-trip: `/tmp/foo.mjs` → `file:///tmp/foo.mjs`, regardless of the host OS.

## Root cause and history

- Introduced by **#3807** (`fix(vscode): convert absolute paths to file:// URLs before import`, merged 2026-04-24).
- That PR only changed files under `vscode/`. The test matrix in `.github/workflows/test.yml` requires `paths: packages/**` (and a few siblings), so the matrix was **not triggered** when #3807 itself was merged — the latent Windows bug slipped through review.
- First exposed on **2026-04-29** when an unrelated v4-base PR (#3788, renovate jsdom) was rebased onto v4 and ran the full matrix.
- Reproduced on 2026-05-04, 2026-05-07, and 2026-05-09 across multiple unrelated v4-base PRs.

CI runs reproducing the failure:

- https://github.com/markuplint/markuplint/actions/runs/25109268035 (2026-04-29, #3788 after rebase)
- https://github.com/markuplint/markuplint/actions/runs/25297742579 (2026-05-04, #3788)
- https://github.com/markuplint/markuplint/actions/runs/25489791893 (2026-05-07, #3788)
- https://github.com/markuplint/markuplint/actions/runs/25593369766 (2026-05-09, #3834)

CI run **before** the regression, confirming the path used to work:

- https://github.com/markuplint/markuplint/actions/runs/24769012803 (2026-04-22, #3788, pre-#3807)

## Impact

- All v4-base PRs that touch `packages/**` are blocked from merging by this Windows-only test failure.
- #3788 (renovate `jsdom@29`) — open since 2026-04-22, blocked.
- #3834 (#3823 astro-parser fix) — blocked.

## Suggested fix direction

1. Have `toImportSpecifier()` short-circuit on a leading `/`: if the input is a POSIX-style absolute path, prepend `file://` directly without going through `pathToFileURL()`.
2. Alternatively, narrow the Windows path detection (e.g. `path.isAbsolute` + drive-letter regex) so `pathToFileURL()` is only called for true Windows-style absolute paths.
3. Add a regression test that mocks `process.platform = 'win32'` so the bug is caught on POSIX CI as well — the existing tests rely on the runtime OS, which is why the bug was never caught locally.

## Process improvement

The `paths:` filter in `test.yml` excludes `vscode/**`. Consider either:

- Adding `vscode/**` to the trigger paths, OR
- Scoping `vitest run` to package paths so vscode tests run in a dedicated vscode workflow.

Either change would have caught this regression at the #3807 review stage.

**Comment by YusukeHirao** [comment]
Fixed by #3841 (merged 2026-05-09 into v4). The dev/v5 mirror was tracked separately as #3842 and fixed by #3843 (merged 2026-05-10), and the related file-resolver bug as #3840 / #3846 (merged 2026-05-10).
